### PR TITLE
Add summary table with conventions and examples

### PR DESCRIPTION
**Pull request by emarino135**
## Description

I created a short summary table for naming conventions.
## Motivation and Context

After a lot of support from my summary cheat sheet I posted on the Godot reddit page
I decided to make this PR to have it included in the docs
image

## How Has This Been Tested?

```
Running Sphinx v4.4.0
loading translations [en]... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] tutorials/scripting/gdscript/gdscript_styleguide     
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorials/scripting/gdscript/index
generating indices... genindex done
writing additional pages... search done
copying downloadable files... [100%] tutorials/scripting/files/autoload.zip    mple/SConstructative.zip
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.
```

### Before addition

image

### After additon

image

### CODE REVIEW

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 1]
```diff
@@ -525,6 +525,20 @@ These naming conventions follow the Godot Engine style. Breaking these will make
...
 
+============ ============= ==================================
+Type         Convention    Example
+============ ============= ==================================
+File names   snake_case    :code:`yaml_parsed.gd`
+Class names   PascalCase   :code:`class_name YAMLParser`
+Node names   PascalCase    :code:`const Weapon = preload("res://weapon.gd")`
+Functions    snake_case    :code:`func load_level():`
+Variables    snake_case    :code:`particle_effect`
+Signals      snake_case    :code:`signal door_opened #always in passed tense`
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:536]:
```suggestion
Signals      snake_case    :code:`signal door_opened`, always in past tense
```

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 2]
```diff
@@ -525,6 +525,20 @@ These naming conventions follow the Godot Engine style. Breaking these will make
...
+Constants    CONSTANT_CASE :code:`const MAX_SPEED = 200`
+Enum names   PascalCase    :code:`enum Element`
+Enum members CONSTANT_CASE :code:`{EARTH, WATER, AIR, FIRE,}`
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:539]: No trailing comma if all values are on a single line:

```suggestion
Enum members CONSTANT_CASE :code:`{EARTH, WATER, AIR, FIRE}`
```

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 3]
```diff
@@ -525,6 +525,20 @@ These naming conventions follow the Godot Engine style. Breaking these will make
 your code clash with the built-in naming conventions, leading to inconsistent
 code.
 
+============ ============= ==================================
+Type         Convention    Example
+============ ============= ==================================
+File names   snake_case    :code:`yaml_parsed.gd`
+Class names   PascalCase   :code:`class_name YAMLParser`
+Node names   PascalCase    :code:`const Weapon = preload("res://weapon.gd")`
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:533]: This is a class name, not a node name.

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 4]
```diff
@@ -525,6 +525,20 @@ These naming conventions follow the Godot Engine style. Breaking these will make
 your code clash with the built-in naming conventions, leading to inconsistent
 code.
 
+============ ============= ==================================
+Type         Convention    Example
+============ ============= ==================================
+File names   snake_case    :code:`yaml_parsed.gd`
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:531]:
```suggestion
File names   snake_case    :code:`yaml_parser.gd`
```

**Review by Calinou — COMMENTED** [review]
Note that we use a different table syntax within the rest of the documentation. I find the one you're using easier to update, but we should try use a consistent style when possible.

**Comment by YuriSizov** [comment]
Shouldn't we add this to the master version as well?

**Comment by Vennnot** [comment]
If only the syntax needs to be changed I can open a separate PR for that

**Comment by tetrapod00** [comment]
In the interest of resolving stalled PRs, I have made https://github.com/godotengine/godot-docs/pull/10295 which retargets this to master, with some tweaks. You are credited as a co-author in the commit.

We should *probably* close this since as a rule we usually don't make many changes to the 3.x docs these days.

**Comment by mhilbrunner** [comment]
Closing as superseded by https://github.com/godotengine/godot-docs/pull/10295, which we just merged. Thanks for contributing!

**Pull request closed without merging**